packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -3298,6 +3298,106 @@ describe('TransactionController Integration', () => {
       transactionController.destroy();
     });
 
+    it('should not block attempts to get the nonce lock for different addresses from the nonceTracker for the networkClientId', async () => {
```

Comment:
Also If we want to be really thorough we could/should do a:

`should not block attempts to get the nonce lock for the same addresses from the nonceTracker for networkClients with different chainIds`

Reply:
added https://github.com/MetaMask/core/pull/3869/commits/d72b3f9d8d92b1083d4f71685402815a245b927c